frontend/packages/console-shared/src/components/dynamic-form/types.ts

```diff
@@ -7,3 +7,23 @@ export enum SchemaType {
   array = 'array',
   object = 'object',
 }
+
+export type DynamicFormFieldOptionsList = {
+  label: string;
+  value: string;
+}[];
+
+export type DynamicFormFieldDependency = {
+  controlFieldPath: string;
+  controlFieldValue: string;
+  controlFieldName: string;
+};
+
+export type UiSchemaOptionsWithDependency = {
+  dependency?: DynamicFormFieldDependency;
+};
+
+export type DynamicFormSchemaError = {
+  title: string;
+  message: string;
+};
```

Comment:
I was having issues with dependency cycles while trying to improve the types for the sort order function in utils, so I moved these reusable types here to prevent future issues.